src/stack.ts

```diff
@@ -176,8 +176,13 @@ export function stack(m: Mark | MarkDef, encoding: Encoding<string>): StackPrope
   if (encoding[dimensionChannel]) {
     const dimensionDef = encoding[dimensionChannel];
     const dimensionField = isFieldDef(dimensionDef) ? vgField(dimensionDef, {}) : undefined;
+    const hasSameDimensionAndStackedField = dimensionField && dimensionField === stackedField;
 
-    if (dimensionField && dimensionField !== stackedField) {
+    // For polar coordinates, do not set a groupBy when working with quantitative fields.
+    const isPolar = isPolarPositionChannel(fieldChannel) || isPolarPositionChannel(dimensionChannel);
+    const shouldAddPolarGroupBy = !isUnbinnedQuantitative(dimensionDef);
+
+    if (isPolar ? shouldAddPolarGroupBy : !hasSameDimensionAndStackedField) {
```

Comment:
I wonder if we really need `isPolar ? shouldAddPolarGroupBy :` clause?   If we strictly check the field above, would that be sufficient?

Reply:
By strictly check, do you mean we'd check if `dimensionDef` and `stackedFieldDef` have the same

- `field` (the current check)
- `bin`
- `aggregator`
- `timeUnit`

for both polar and cartesian graphs?

Currently if I just remove the `isPolar ? shouldAddPolarGroupBy` clause and use the existing check, we'd introduce [this bug](https://github.com/vega/vega-lite/pull/9537/commits/c2de794c127f33c19ccec6e47adfb442c13c98a7)

Reply:
Made a PoC commit to see if this is what you had in mind

https://github.com/vega/vega-lite/pull/9537/commits/8b010a61aed5a455d5e60e52ec9ddb1c18323c63

Reply:
>we'd introduce this bug

can't find this link

Reply:
Fixed link: https://github.com/vega/vega-lite/blob/231ed17c2b37811701a89029ab1524750abdd80b/examples/compiled/arc_quantitative_stacked_theta.png

Reply:
I found it interesting to see how many of the visual tests break if I removed the check that only applies this change to polar encodings.  

https://github.com/vega/vega-lite/pull/9557